hrm-service/src/permissions/actions.ts

```diff
@@ -103,26 +107,64 @@ const isAddIncident = (change: Diff<any>) =>
 const isAddDocument = (change: Diff<any>) =>
   isNewOrAddKind(change) && isPathTarget(['info', 'documents'])(change.path);
 
-const isEditNote = (change: Diff<any>) =>
-  isEditKind(change) && isPathTarget(['info', 'notes'])(change.path);
+const isSectionEdit = (sectionName: string) => (change: Diff<any>) =>
+  (isEditKind(change) && isPathTarget(['info', sectionName])(change.path)) ||
+  isDescendantPathTarget(['info', sectionName])(change.path);
```

Comment:
Looks like one of my question didn't went with the review so I'm re-posting it.
I'm probably wrong but looks to me like this will evaluate to `true` if
  - the change is an edit kind and is an exact match in the path.
  - the change is a descendant of the info -> sectionName object.
Does this means that `isSectionEdit` will evaluate to `true` if we are adding a new entry to the info -> sectionName? If so, should that be recognized as an edit, since we have separate rules for addition?

Reply:
So I _think_, based on what the tests are showing that:

Adding a new item to the household collection (for example) is an 'A' change on ['info', 'household'] - which is picked up as an 'Add'

Editing the 'myProp' property of the second item of the household collection is an 'E' change on ['info', 'household', 1, ''myProp'] - caught by the 'descendant path' check as an 'Edit'.

Adding the 'myProp' property of the second item of the household collection is an 'N' change on ['info', 'household', 1, ''myProp'] - caught by the 'descendant path' check as an 'Edit'.

Removing the 'myProp' property of the second item of the household collection is an 'D' change on ['info', 'household', 1, ''myProp'] - caught by the 'descendant path' check as an 'Edit'.

Also presumable adding items to the 'myProp' array property of the second item of the household collection would be an 'A' change on ['info', 'household', 1, ''myProp'] - caught by the 'descendant path' check, but I don't think we have any of those.

The first part of the check - i.e. `isEditKind(change) && isPathTarget(['info', sectionName])(change.path))` is probably redundant, we wouldn't see 'edit' changes directly on ['info', 'household'] during normal operation.

If you think there are inconsistencies in the logic, review the service tests I created and if there are cases you think they don't cover which might not behave correctly, let me know and I'll add a test case. There are literally 1000s of possible edit scenarios and we couldn't test them all, but we do want to be sure we are covering the more likely problem scenarios

Reply:
No I don't see any kind of inconsistencies. I was just confused on why there won't be any overlaps on `isAddSection` and `isEditSection`. I think after another pass thru the code, I get it, it depends on the exact path match vs the descendant path match, not only the change kind. Thanks!

PS: only a maniac will dare to defy so many tests! xD